Source/GSCSConstraint.m

```diff
@@ -1,27 +1,366 @@
+                                      strength: [GSCSStrength strengthRequired]
+                                      variable: nil];
+}
+
+- (instancetype) initLinearInequityConstraintWithExpression:
+    (GSCSLinearExpression *)expression
+{
+  return
+      [self initWithType: GSCSConstraintTypeLinearInequity
+                strength: [GSCSStrength strengthRequired]
+              expression: expression
+                variable: nil];
+}
+
+- (instancetype) initEditConstraintWithVariable: (GSCSVariable *)variable
+                                        stength: (GSCSStrength *)strength
+{
+  GSCSLinearExpression *expression =
+      [[GSCSLinearExpression alloc] initWithVariable: variable
+                                         coefficient: -1
+                                            constant: [variable value]];
+  GSCSConstraint *constraint = [self initWithType: GSCSConstraintTypeEdit
+                   strength: strength
+                 expression: expression
+                   variable: variable];
+  RELEASE(expression);
+  return constraint;
+}
+
+- (instancetype) initStayConstraintWithVariable: (GSCSVariable *)variable
+                                       strength: (GSCSStrength *)strength
+{
+  GSCSLinearExpression *expression =
+      [[GSCSLinearExpression alloc] initWithVariable: variable
+                                         coefficient: -1
+                                            constant: [variable value]];
+  GSCSConstraint *constraint = [[GSCSConstraint alloc] initWithType: GSCSConstraintTypeStay
+                                     strength: strength
+                                   expression: expression
+                                     variable: variable];
+  RELEASE(expression);
+  return constraint;
+}
+
+- (instancetype) initWithLhsVariable: (GSCSVariable *)lhs
+                   equalsRhsVariable: (GSCSVariable *)rhs
+{
+  GSCSLinearExpression *expression =
+      [[GSCSLinearExpression alloc] initWithVariable: lhs];
+  [expression addVariable: rhs coefficient: -1];
+  GSCSConstraint *constraint = [self initLinearConstraintWithExpression: expression];
+  RELEASE(expression);
+
+  return constraint;
+}
+
+- (instancetype) initWithLhsVariable: (GSCSVariable *)lhs
+                      equalsConstant: (CGFloat)rhs
+{
+  GSCSLinearExpression *expression =
+      [[GSCSLinearExpression alloc] initWithVariable: lhs
+                                         coefficient: 1
+                                            constant: -rhs];
+  GSCSConstraint *constraint = [self initLinearConstraintWithExpression: expression];
+  RELEASE(expression);
+  return constraint;
+}
+
++ (GSCSConstraint *) constraintWithLeftVariable: (GSCSVariable *)lhs
+operator: (GSCSConstraintOperator) operator
+                                  rightVariable: (GSCSVariable *)rhsVariable
+{
+  if (operator== GSCSConstraintOperatorEqual)
+    {
+      GSCSLinearExpression *expression =
```

Comment:
I would use the above method `-initWithLhsVariable:equalsRhsVariable:`here to reduce the amount of code.